applications/FluidDynamicsApplication/custom_elements/embedded_navier_stokes.h

```diff
@@ -277,215 +246,146 @@ class EmbeddedNavierStokes : public NavierStokes<TDim, TNumNodes>
      */
     virtual int Check(const ProcessInfo& rCurrentProcessInfo) override
     {
-        KRATOS_TRY
+        KRATOS_TRY;
 
         // Base element check
         int ErrorCode = NavierStokes<TDim, TNumNodes>::Check(rCurrentProcessInfo);
         if(ErrorCode != 0) return ErrorCode;
 
         // Specific embedded element check
         if(DISTANCE.Key() == 0)
-            KRATOS_THROW_ERROR(std::invalid_argument,"DISTANCE Key is 0. Check if the application was correctly registered.","");
+            KRATOS_ERROR << "DISTANCE Key is 0. Check if the application was correctly registered.";
 
```

Comment:
You can also check that DISTANCE is assigned to the nodal database (SolutionStepDataHas())